### PR TITLE
Vmal

### PR DESCRIPTION
**Pull request by Malisevych**

### CODE REVIEW

File: MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/ProductPageController.java [hunk 1]
```diff
@@ -213,19 +198,124 @@ public String delProduct(@RequestParam int productId){
...
+        Product product = restTemplate.getForObject(getProductUri, Product.class, param);
+
+        param.clear();
+        param.put("productId", productId);
+        param.put("userId", userId);
 
-        RestTemplate restTemplate = new RestTemplate();
-        Store[] result = restTemplate.getForObject(uri, Store[].class, param);
-        List<Store> stores = Arrays.asList(result);
+        Store[] arrStores = restTemplate.getForObject(getStoresUri, Store[].class, param);
```
Comment (kucherif) [inline MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/ProductPageController.java]: Please try this one to resolve issue with array and list
`ResponseEntity<List<Rate>> rateResponse =
        restTemplate.exchange("https://bitpay.com/api/rates",
                    HttpMethod.GET, null, new ParameterizedTypeReference<List<Rate>>() {
            });
List<Rate> rates = rateResponse.getBody();`

File: RESTserver/src/main/java/com/softserve/if072/restservice/service/ProductService.java [hunk 2]
```diff
@@ -64,6 +65,9 @@ public Product getProductById(int id) throws DataNotFoundException {
     public void updateProduct(Product product) throws DataNotFoundException {productDAO.update(product);}
 
     @Transactional
+    public void updateProductByImage(Product product) throws DataNotFoundException {productDAO.updateImage(product);}
```
Comment (kucherif) [inline RESTserver/src/main/java/com/softserve/if072/restservice/service/ProductService.java]: format the code

File: MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/ProductPageController.java [hunk 3]
```diff
@@ -213,19 +198,124 @@ public String delProduct(@RequestParam int productId){
...
+                for(Store ns : newStores) {
+                    if(os.getId() == ns.getId()) {
+                        continue outer;
+                    }
+                }
+                storesToDelete.add(os);
+            }
+        } else {
+            storesToDelete.addAll(oldStores);
+        }
```
Comment (kucherif) [inline MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/ProductPageController.java]: How I would rewrite this loops:
```
List<Store> storesToAdd = new ArrayList<>(newStores);
List<Store> storesToDelete = new ArrayList<>(oldStores);

storesToAdd.removeAll(oldStores);
storesToDelete.removeAll(newStores);
```
Does it make sens? Is it easier to understand?

However, for this, you need to have proper equals method implemented in the Store class, or you can do this with IDs and then get real Store object, what can reflect on performance, less amount of data to send over the network.

Comment (Malisevych) [inline MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/ProductPageController.java]: It's good idea! Have done.

Comment (Malisevych) [inline MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/ProductPageController.java]: Thanks

File: MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/ProductPageController.java (hunk 1 above)
Comment (Malisevych) [inline MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/ProductPageController.java]: It works. Thanks

**Pull request merged**